File: CCC-school-work/comporg/pnc2/abratkov/js/primeregbs.js
//
// Primeregbs.js - Andrei Bratkovski
//

//
// Process arguments based on the 1st command line
// argument
//
var maxNumPrimes = process.argv[2];

//
// Declare counter variables
//
var increment = 0;
var lowerBound = 2;
var numPrimes = 0;
var compositeCounter = 0;

//
// Start timer
//
var start = new Date().getTime();

//
// Calculate primes by checking up to the square root of current number being
// checked, until number of primes is the same as the chosen number of max primes
//
while ( numPrimes < maxNumPrimes )
{
  for ( increment = 2; increment <= Math.sqrt(lowerBound); increment++)
  {
    if ( lowerBound % increment == 0 )
      compositeCounter++;
      
    if ( compositeCounter >= 2)
      break;
  }
  if ( compositeCounter < 1 )
  {
      process.stderr.write(lowerBound + " ")
      numPrimes++;
  }
   
  lowerBound++;
  compositeCounter = 0;
}
//
// Calculate end time and calculate
// how long program takes
//
console.log("");
var end = new Date().getTime();
console.log((end-start)/1000);

File: CCC-school-work/comporg/pnc1/ruby/primeregbs.rb
#!/usr/bin/ruby -w 
# primeregbs.rb - Andrei

#
# Declare variables, and start the timer.
# the max prime number is our first command line
# argument
#
lowerBound = 2
numPrimes = 0

numMaxPrimes = ARGV[0].to_i
compositeCounter = 0
start = Time.now

#
# Iterate over each number up to its square root
# and check if it is prime, adding to a compositeCounter
# to keep track of its primality
#
while (numPrimes < numMaxPrimes) do
	increment = 2
	while (increment <= Math.sqrt(lowerBound)) do
		if (lowerBound % increment == 0)
			compositeCounter += 1
		end
		
		if (compositeCounter >= 2)
			break
		end
		increment += 1
	end
	if (compositeCounter < 1)
		$stderr.print "#{lowerBound} "
		numPrimes += 1
	end
	lowerBound += 1
	compositeCounter = 0
end

#
# Print the finishing time of
# the program
#
finish = Time.now
diff = finish - start
puts "\n#{diff}"
